remotespark/livyclientlib/livysession.py

```diff
@@ -106,11 +107,38 @@ def create_sql_context(self):
         else:
             raise FailedToCreateSqlContextException("Failed to create the SqlContext.\nError: '{}'".format(out))
 
+    def get_app_id(self):
+        if self._app_id is None:
+            session = self._http_client.get_session(self.id)
+            self._app_id = session['appId']
+        return self._app_id
+
+    def get_app_info(self):
+        session = self._http_client.get_session(self.id)
+        if 'appInfo' in session:
+            return session['appInfo']
+        else:
+            return {}
+
+    def get_app_info_member(self, member_name, default_value):
+        app_info = self.get_app_info()
+
+        if member_name in app_info and app_info[member_name] is not None:
+            return app_info[member_name]
+        else:
+            return default_value
+
+    def get_driver_log_url(self):
+        return self.get_app_info_member("driverLogUrl", "No driver log URL is available yet.")
```

Comment:
The message saying that the url is not available yet is misleading for versions of Livy that do not expose the URL at all...